lib/modules/dosomething/dosomething_api/resources/campaign_resource.inc

```diff
@@ -359,6 +362,23 @@ function _campaign_resource_reportback($nid, $values) {
     watchdog('dosomething_api', '_campaign_resource_reportback:' . json_encode($values));
   }
 
-  return dosomething_reportback_save($values, $user);
+  $transaction_id = $_SERVER['HTTP_X_REQUEST_ID'];
+
+  if (! $transaction_id) {
```

Comment:
Yes! This is to prevent the infinite loop. 

So when the mobile app or sms sends a reportback, it won't have a transaction id. Since it doesn't have one, we know it comes from those two sources and we want to send to Rogue. 

Then, Rogue receives it and sends back to Phoenix with this same API. It will now have a transaction id so we don't want it to send to Rogue again and get stuck in an infinite loop. So instead of sending back to Rogue, we just save it in Phoenix.